include/alpaka/core/Common.hpp

```diff
@@ -23,20 +23,75 @@
 
 #include <alpaka/core/Debug.hpp>
 
-#include <type_traits>                      // std::enable_if
+#include <boost/predef/version_number.h>   // BOOST_VERSION_NUMBER
 
-#if (!defined(__CUDA_ARCH__)) || defined(NDEBUG)
-    #include <boost/core/ignore_unused.hpp> // boost::ignore_unused
+//-----------------------------------------------------------------------------
+// BOOST_PREDEF_MAKE_10_VVRRP(V)
+//-----------------------------------------------------------------------------
+#define BOOST_PREDEF_MAKE_10_VVRRP(V) BOOST_VERSION_NUMBER(((V)/1000)%100,((V)/10)%100,(V)%10)
+
+//-----------------------------------------------------------------------------
+// CUDA language detection
+// - clang defines __CUDA__ when compiling CUDA code ('-x cuda')
+// - nvcc defines __CUDACC__ when compiling CUDA code
+//-----------------------------------------------------------------------------
+#if defined(__CUDA__) || defined(__CUDACC__)
+    #include <cuda.h>
+    #define BOOST_LANG_CUDA BOOST_PREDEF_MAKE_10_VVRRP(CUDA_VERSION)
```

Comment:
I am exending the Boost Predef library here. To make it more clear that the values of `BOOST_LANG_CUDA` and the other macros follow the rules of the Predef library, I use the same 'macro namespace'. Finally, if they prove beneficial, I intent to propose those extensions to Boost Predef itself.